prober/utils.go

```diff
@@ -119,3 +122,14 @@ func ipHash(ip net.IP) float64 {
 	h.Write(ip)
 	return float64(h.Sum32())
 }
+
+func internationalizeDNSDomain(logger log.Logger, domain string) string {
+	idnaDomain, err := idna.Lookup.ToASCII(domain)
+	if err != nil {
```

Comment:
I don't think we should be silently ignoring errors.

Reply:
Somehow we pass :: in some tests, which creates an error disallowed rune U+003A. I need to investigate why, but I was thinking that it would be safe to ignore those errors.

Reply:
I see what's going on here.

You have to try to handle the hostname as an IP address (either IPv4 or IPv6), `net.ParseIP` will do this for you, and if that fails handle it as a hostname and apply a punycode transformation to it.

Basically:

```
if net.ParseIP(domain) != nil {
    return domain
}

idnaDomain, err := idna.Lookup.ToASCII(domain)
...
```

Reply:
Thanks